lib/class-factory.js

```diff
@@ -669,9 +669,13 @@ function ClassFactory (vm) {
           try {
             const fieldName = invokeObjectMethodNoArgs(env.handle, field, fieldGetName);
             try {
-              const fieldjsName = env.stringFromJni(fieldName);
+              var fieldjsName = env.stringFromJni(fieldName);
               const fieldHandle = env.newGlobalRef(field);
               fieldHandles.push(fieldHandle);
+              // If we have a collided method, suffix the fieldName
+              if(jsMethods.hasOwnProperty(fieldjsName)){
```

Comment:
- Comment can be removed as it's already clear what the code is doing.
- Missing space after `if`.
- Theoretically there might be a method prefixed with `_` having the same name, so using `while` here would be better.